kernel/include/syscall_handler.h

```diff
@@ -406,12 +406,12 @@ static inline int _obj_validation_check(struct _k_object *ko,
 	ARG_UNUSED(obj);
 #endif
 
-	return ret;
+	return ret == 0 ? true : false;
 }
 
 #define Z_SYSCALL_IS_OBJ(ptr, type, init) \
 	Z_SYSCALL_VERIFY_MSG( \
-	    !_obj_validation_check(_k_object_find((void *)ptr), (void *)ptr, \
+	    _obj_validation_check(_k_object_find((void *)ptr), (void *)ptr, \
```

Comment:
that's a bit hard to read, now. I wonder if the function name could also be updated to reflect the revert of the return value